data/network/src/commonMain/kotlin/social/androiddev/common/network/MastodonApiKtor.kt

```diff
@@ -15,11 +15,14 @@ import io.ktor.client.plugins.ResponseException
 import io.ktor.client.request.forms.formData
 import io.ktor.client.request.get
 import io.ktor.client.request.post
+import io.ktor.http.URLProtocol
+import io.ktor.http.path
 import kotlinx.serialization.SerializationException
 import social.androiddev.common.network.model.Application
 import social.androiddev.common.network.model.Instance
 import social.androiddev.common.network.util.runCatchingIgnoreCancelled
-class MastodonApiImpl(
+
+internal class MastodonApiKtor(
```

Comment:
FTR I think `Impl` was likley fine as a name, as long as the class isn't leaking out of the module. You could also call it `RealMastodonApi`. Depends whether we want dependencies to be called out in class names. In this case maybe Ktor is ok, since we're injecting it with a class that comes from Ktor.

Reply:
Yeah I thought to myself the suffix Ktor gives more value in naming than `Impl`, in that this client implementation is heavily coupled with Ktor. And if we wanted to swap out the client implementation with something else, it would become more clear than just recreating another "Impl" at that time and struggling with naming.